packages/runtime/container-runtime/src/id-compressor/README.md

```diff
@@ -1,3 +1,69 @@
-# ID Compressor
+# **Distributed ID Allocator**
 
-An [IdCompressor](./idCompressor.ts) generates unique identifiers that may be efficiently compressed/decompressed and synchronized with other IdCompressors.
+A library which generates small number representations of arbitrary non-colliding Version 4 UUIDs (stable IDs) across multiple sessions in a network. This scheme enables a distributed application to utilize the global uniqueness guarantees of UUIDs while maintaining the performance advantages of small integers.
+
+## **Overview**
+
+The distributed ID allocation scheme allows clients to use, store, and reference small numbers that map to UUIDs. The primary benefits are improved memory efficiency and reduced storage size, as well as improved runtime performance when used as keys in collections or other data structures in place of UUIDs.
+
+The scheme requires a total order for ID allocation, so the allocator is designed to work within a service that provides centralized total order broadcast.
+
+## **Sessions**
+
+A session can be thought of as a unique identifier for a single allocator. Even if an allocator is taken offline, serialized to disk, and rehydrated back, it's considered the same session if it has the same session ID.
+
+## **Document**
+
+The set of IDs generated by all Sessions contributing to the same distributed collection (e.g. an individual collaborative document). Uniqueness of allocated IDs is limited to the document context (unless in their UUID form), and thus IDs in their compressed form are not portable between documents.
+
+## **Generation API**
+
+New IDs are generated via synchronous API on the allocator. The API returns a small number representation of the UUID with the following guarantees:
```

Comment:
```suggestion
New IDs are generated via a synchronous API on the allocator. The API returns a small number representation of the UUID with the following guarantees:
```
@alex-pardes Asynchronous collaboration for a synchronous collaboration environment 12/11/17